### PR TITLE
HAAR-667: Reset password for suspended user.

### PR DESCRIPTION
**Pull request by ravmoj**
Added staff status to UserPersonDetail and UserDetail

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/jpa/StaffStatus.kt [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package uk.gov.justice.digital.hmpps.nomisuserrolesapi.jpa
+
+enum class StaffStatus(
+  val code: Int,
+  val desc: String,
+) {
+  Active(0, "Active"),
```
Comment (karenmillermoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/jpa/StaffStatus.kt:7]: The Active enum be in Capitals too

Comment (karenmillermoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/jpa/StaffStatus.kt:7]: Bit confused with the Active and ACTIVE enums

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/data/UserDetail.kt [hunk 2]
```diff
@@ -16,14 +17,23 @@ data class UserDetail(
...
+    example = "test@test.com",
+    required = false
+  ) val primaryEmail: String?,
   @Schema(description = "List of associated DPS Role Codes", required = false) val dpsRoleCodes: List<String>,
   @Schema(description = "Account is not locked", required = false) val accountNonLocked: Boolean?,
   @Schema(description = "Credentials are not expired flag", required = false) val credentialsNonExpired: Boolean?,
   @Schema(description = "User is enabled flag", required = true) val enabled: Boolean,
   @Schema(description = "User is admin flag", required = false) val admin: Boolean?,
   @Schema(description = "User is active flag", required = true) val active: Boolean,
+  @Schema(description = "Staff Status", example = "ACTIVE", required = true) val staffStatus: StaffStatus?,
```
Comment (karenmillermoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/data/UserDetail.kt:36]: If required = true should this not be nullable?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/jpa/StaffStatus.kt (hunk 1 above)
Comment (ravmoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/jpa/StaffStatus.kt:7]: Active and ACTIVE both are status in Nomis. Just taken all statuses.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/data/UserDetail.kt (hunk 2 above)
Comment (ravmoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/data/UserDetail.kt:36]: thanks Karen..changed

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/jpa/StaffStatus.kt (hunk 1 above)
Comment (Mjwillis) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/jpa/StaffStatus.kt:7]: @ravmoj need to check PROD -as not always the same as Dev

Comment (Mjwillis) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/jpa/StaffStatus.kt:7]: Suggest if only in Dev we update the data

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/jpa/UserPersonDetail.kt [hunk 3]
```diff
@@ -183,13 +190,14 @@ data class UserPersonDetail(
...
 
   private fun addUserGroup(userGroup: UserGroup) {
-    activeAndInactiveMemberOfUserGroups.firstOrNull { it.id.userGroupCode == userGroup.id && it.id.username == this.username } ?: run {
-      val member = UserGroupMember(
-        id = UserGroupMemberPk(userGroupCode = userGroup.id, username = this.username),
-        user = this, userGroup = userGroup
-      )
-      activeAndInactiveMemberOfUserGroups.add(member)
-    }
+    activeAndInactiveMemberOfUserGroups.firstOrNull { it.id.userGroupCode == userGroup.id && it.id.username == this.username }
```
Comment (Mjwillis) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/jpa/UserPersonDetail.kt:193]: This is confusing as these are not actual changes

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/resource/UserResourceTest.kt [hunk 4]
```diff
@@ -53,7 +55,8 @@ class UserResourceTest {
       activeCaseload = PrisonCaseload("MDI", "Moorland"),
       dpsRoleCount = 3,
       email = "joe@bloggs.com",
-      status = AccountStatus.OPEN
+      status = AccountStatus.OPEN,
+      staffStatus = StaffStatus.ACTIVE
```
Comment (Mjwillis) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/resource/UserResourceTest.kt:59]: What happens if we add more status's to this data? will it fall over?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/jpa/StaffStatus.kt (hunk 1 above)
Comment (Mjwillis) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/jpa/StaffStatus.kt:7]: Think this is reference data - can't really use Enum

**Review by Mjwillis — COMMENTED** [review]
Some comments

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/jpa/UserPersonDetail.kt (hunk 3 above)
Comment (ravmoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/jpa/UserPersonDetail.kt:193]: Format change

**Review by Mjwillis — APPROVED** [review]
LGTM

**Pull request merged**